ProjFS.Mac/PrjFSLib/prjfs-log/prjfs-log.cpp

```diff
@@ -11,6 +11,31 @@ static uint64_t nanosecondsFromAbsoluteTime(uint64_t machAbsoluteTime);
 
 static mach_timebase_info_data_t s_machTimebase;
 
+static dispatch_source_t start_kext_profiling_data_polling(io_connect_t connection);
+static bool fetch_kext_profiling_data(io_connect_t connection);
+
+#define PCNAME(SUFFIX) [Probe_ ## SUFFIX] = #SUFFIX
+static const char* const PerfCounterNames[Probe_Count] =
+{
+    PCNAME(VnodeOp),
```

Comment:
Note that the macro doesn't just expand the name, it also generates the assignment, so the idea is you can't accidentally mismatch enum value and name string.

Reply:
I guess this is just a style difference. I'm pretty allergic to macros that mangle symbols because they tend to make it easier to write code at the expense of making it harder to read code. In this case, the biggest problem I have with it is I can't put my cursor on the enum value and jump to its definition. It's not even obvious that it _is_ an enum value, so I wouldn't even know where to go find it.